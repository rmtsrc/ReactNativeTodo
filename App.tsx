import React from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import Constants from 'expo-constants';

import AddTodo from './components/AddItem';
import TodoList from './components/TodoList';
import { TodoStoreProvider } from './modal/TodoStore';

export default function App() {
  return (
    <SafeAreaView style={styles.container}>
      <TodoStoreProvider>
        <AddTodo />
        <TodoList />
      </TodoStoreProvider>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: Constants.statusBarHeight + 10,
  },
});
