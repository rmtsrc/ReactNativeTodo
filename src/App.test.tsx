import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

describe('<App />', () => {
  const tree = renderer.create(<App />).toJSON();

  test('renders correctly without error', () => {
    expect(tree).toMatchSnapshot();
  });
});
