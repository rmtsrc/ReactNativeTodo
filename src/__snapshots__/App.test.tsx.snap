// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly without error 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
      "marginTop": 54,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomWidth": 1,
          "borderColor": "#86939e",
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 15,
            "paddingRight": 20,
          }
        }
      >
        <View>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <TextInput
        allowFontScaling={true}
        editable={true}
        onChangeText={[Function]}
        onSubmitEditing={[Function]}
        placeholder="What needs to be done?"
        rejectResponderTermination={true}
        style={
          Object {
            "alignSelf": "center",
            "color": "black",
            "flex": 1,
            "fontSize": 18,
            "minHeight": 40,
          }
        }
        testID="RNE__Input__text-input"
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
